test/e2e/cli/cli_console_test.go

```diff
@@ -260,3 +224,68 @@ var _ = Describe("CLI - device console", Serial, func() {
 func eventuallySlow(actual any) types.AsyncAssertion {
 	return Eventually(actual).WithTimeout(LONG_TIMEOUT).WithPolling(LONG_POLLING)
 }
+
+// -----------------------------------------------------------------------------
+// Helper functions
+// -----------------------------------------------------------------------------
+
+// extractTimestampsFromLogs extracts and parses timestamps from log lines that match the given regex pattern.
+// Returns a slice of valid timestamps found in the logs.
+func extractTimestampsFromLogs(logs string, logPattern *regexp.Regexp) []time.Time {
+	lines := strings.Split(strings.TrimSpace(logs), "\n")
+	logrus.Infof("Read %d log lines from agent journal", len(lines))
+
+	var validTimestamps []time.Time
+
+	for _, line := range lines {
+		if m := logPattern.FindStringSubmatch(line); m != nil {
+			if t, err := time.Parse(time.RFC3339Nano, m[1]); err == nil {
```

Comment:
nit: this is suble but its best practice to evaluate the error first in go so just switch these conditions

```suggestion
			if t, err := time.Parse(time.RFC3339Nano, m[1]); err != nil {
```